### DISCUSSION TITLE
Quality of stream

### DISCUSSION THREAD

**Discussion by Cornmeister** [bluenviron/mediamtx#2254] [Questions]
Hi all,

I've built a bird feeder with camera, Pi4 with a Raspcamv3 (wide),
everything is working fine, there is one thing I cant fix (yet).

When a bird enters the feeder, the quality of the stream drops,
see this youtube movie as an example:
https://youtu.be/U_Q4YzuF4Bk

Any idea what causes this ?
Mediamtx ? or is the pi not strong enough ?

**Comment by aler9** [comment]
Hello,
First of all, that's quite an interesting application of the camera.

Video quality depends on the video encoder, which is placed in the Raspberry hardware, not in the server, and is controlled by the `rpiCamera*` parameters in the configuration file. In particular, the `rpiCameraBitrate` parameter specifies the maximum bitrate. Modern video codecs like H264 works by detecting changes between the current frame and the previous one, and they save the changes only. Consequently, if nothing changes, the video stream is smaller in terms of bitrate, while when something changes, more bandwidth is required.

What happens is that when the bird arrives, the resulting stream gets bigger than  `rpiCameraBitrate`, and the encoder is forced to decrease the quality in order to stay below  it.

Increase `rpiCameraBitrate`.

**Reply by Cornmeister** [reply]
Ow wow, tnx for the explanation.
Gonna try it asap!

[a comment by an automated account is not shown]
